river/ensemble/boosting.py

```diff
@@ -118,6 +119,98 @@ def predict_proba_one(self, x):
+    with the addition of the `ADWIN` algorithm as a change detector. If concept
+    drift is detected, the worst member of the ensemble (based on the error
+    estimation by ADWIN) is replaced by a new (empty) classifier.
+    Parameters
+    ----------
+    model
+        The classifier to boost.
+    n_models
+        The number of models in the ensemble.
+    seed
+        Random number generator seed for reproducibility.
+    Examples
+    --------
+    >>> from river import datasets
+    >>> from river import ensemble
+    >>> from river import evaluate
+    >>> from river import linear_model
+    >>> from river import metrics
+    >>> from river import preprocessing
+    >>> dataset = datasets.Phishing()
+    >>> model = ensemble.ADWINBoostingClassifier(
+    ...     model=(
+    ...         preprocessing.StandardScaler() |
+    ...         linear_model.LogisticRegression()
+    ...     ),
+    ...     n_models=3,
+    ...     seed=42
+    ... )
+    >>> metric = metrics.F1()
+    >>> evaluate.progressive_val_score(dataset, model, metric)
+    F1: 87.41%
+
+    References
+    ----------
+    [^1]: Albert Bifet, Geoff Holmes, Bernhard Pfahringer, Richard Kirkby,
+    and Ricard Gavaldà. "New ensemble methods for evolving data streams."
+    In 15th ACM SIGKDD International Conference on Knowledge Discovery and
+    Data Mining, 2009.
+    [^2]: Oza, N., Russell, S. "Online bagging and boosting."
+    In: Artificial Intelligence and Statistics 2001, pp. 105–112.
+    Morgan Kaufmann, 2001.
+    """
+
+    def __init__(self, model: base.Classifier, n_models=10, seed: int = None):
+        super().__init__(model, n_models, seed)
+        self._drift_detectors = [copy.deepcopy(drift.ADWIN()) for _ in range(self.n_models)]
+
+    def learn_one(self, x, y):
+
+        change_detected = False
+        lambda_poisson = 1.0
+        for i, model in enumerate(self):
+            for _ in range(utils.random.poisson(1, self._rng)):
+                model.learn_one(x, y)
+
+            if model.predict_one(x) == y:
+                self.correct_weight[i] += lambda_poisson
+                lambda_poisson *= (self.correct_weight[i] + self.wrong_weight[i]) / (
+                    2 * self.correct_weight[i]
+                )
+
+            else:
+                self.wrong_weight[i] += lambda_poisson
+                lambda_poisson *= (self.correct_weight[i] + self.wrong_weight[i]) / (
+                    2 * self.wrong_weight[i]
+                )
+
+            try:
+                y_pred = model.predict_one(x)
+                error_estimation = self._drift_detectors[i].estimation
+                self._drift_detectors[i].update(int(y_pred == y))
+                if self._drift_detectors[i].drift_detected:
+                    if self._drift_detectors[i].estimation > error_estimation:
+                        change_detected = True
+            except ValueError:
+                change_detected = False
```

Comment:
Can you comment where the `ValueError` is raised? I always find it a bit hacky to rely on an exception

Reply:
This part of the code was based in the ADWINBagging code. I didn't find out what raise the ValueError...